sway/input/seat.c

```diff
@@ -402,6 +402,22 @@ static void drag_icon_handle_destroy(struct wl_listener *listener, void *data) {
 	free(icon);
 }
 
+static void drag_handle_destroy(struct wl_listener *listener, void *data) {
+	struct sway_drag *drag = wl_container_of(listener, drag, destroy);
+
+	// Focus doesn't update during the drag, so focus the drag target
+	if (drag->wlr_drag->focus) {
+		struct sway_view *focused = view_from_wlr_surface(drag->wlr_drag->focus);
+		if (focused != NULL && focused->container != NULL) {
+			seat_send_focus(&focused->container->node, drag->seat);
```

Comment:
Can a drag end with a layer surface or an Xwayland unmanaged surface as the target? It sounds to me like they can, and if that's the case this should have [some more logic](https://github.com/swaywm/sway/blob/b7a6cc506a952cfa20918ddc9304e580a0371b05/sway/input/keyboard.c#L556) for those cases (since they won't have `sway_view`s associated).

Reply:
Yeah, good point. Ended up changing the logic to just mimic the keyboard leave logic. Just refocusing what is already focused is sufficient, and better than overriding focus logic locally.